File: Homework5/myreplaceall.erl
-module(myreplaceall).
-export([myreplaceall/3]).

myreplaceall(_,_,[])  -> [];
myreplaceall(X,Y,[H|T]) ->
    [ if H =:= Y -> X;
         true -> H
    end | myreplaceall(X,Y,T)].

%X = replace
%Y = Find

File: Homework3/myreplaceall.hs
myreplaceall :: (Eq a) => a -> a -> [a] -> [a]
myreplaceall replace find list
    | null list         = []
    | find == head list = myreplaceall replace find (replace : tail list)
    | otherwise         = head list : myreplaceall replace find (tail list)

myreplaceall_pm :: (Eq a) => a -> a -> [a] -> [a]
myreplaceall_pm replace find [] = []
myreplaceall_pm replace find (x : xs)
    | find == x = myreplaceall_pm replace find (replace : xs)
    | otherwise = x : myreplaceall replace find xs
